calendar/src/main/java/com/toggl/calendar/calendarday/ui/views/CalendarItemDrawingDelegate.kt

```diff
@@ -1,21 +1,337 @@
+    private val calendarIconSize: Float,
+    private val calendarIconRightInsetMargin: Float,
+    private val shortCalendarItemHeight: Float,
+    private val shortCalendarItemFontSize: Float,
+    private val regularCalendarItemFontSize: Float,
+    private val shortCalendarItemVerticalPadding: Float,
+    private val regularCalendarItemVerticalPadding: Float,
+    private val shortCalendarItemHorizontalPadding: Float,
+    private val regularCalendarItemHorizontalPadding: Float,
+    private val primaryTextColor: Int
+) {
+    private val textLayouts: MutableMap<String, StaticLayout> = mutableMapOf()
+    private val itemPaint: Paint = Paint(Paint.ANTI_ALIAS_FLAG)
+    private val textPaint: Paint = Paint(Paint.ANTI_ALIAS_FLAG)
+    private val calendarIconPaint: Paint = Paint(Paint.ANTI_ALIAS_FLAG)
     private val drawingRect: RectF = RectF()
+    private val stripeRect = RectF()
+    private val dashEffect = DashPathEffect(floatArrayOf(10f, 10f), 0f)
+    private val minimumTextContrast = 1.6f
+
+    private var defaultProjectColor = "#c2c2c2"
+    private var minHourHeight: Float = 0f
+    var currentHourHeight: Float = 0f
+        set(value) {
+            minHourHeight = value / 4
+            field = value
+        }
 
     fun onDraw(canvas: Canvas, calendarBounds: RectF, calendarItem: CalendarItem) {
-        calendarItem.run {
-            drawingRect.set(calendarBounds.left, calendarBounds.top, calendarBounds.right, 300f)
-            canvas.drawRect(drawingRect, itemPaint)
+        calculateItemRect(calendarItem, calendarBounds)
+        when {
+            calendarItem is CalendarItem.TimeEntry && calendarItem.duration() != null ->
+                drawStoppedTimeEntryItem(canvas, calendarItem)
+            calendarItem is CalendarItem.TimeEntry && calendarItem.duration() == null ->
+                drawRunningTimeEntryItem(canvas, calendarItem)
+            calendarItem is CalendarItem.CalendarEvent -> drawCalendarEventItem(canvas, calendarItem)
         }
+        drawCalendarItemText(canvas, calendarItem)
+    }
+
+    private fun drawCalendarItemText(canvas: Canvas, calendarItem: CalendarItem) {
+        val textLeftPadding =
+            if (calendarItem is CalendarItem.CalendarEvent) calendarIconSize - calendarIconRightInsetMargin else 0f
+        val eventHeight = drawingRect.height()
+        val eventWidth = drawingRect.width() - textLeftPadding
+        val fontSize =
+            if (eventHeight <= shortCalendarItemHeight) shortCalendarItemFontSize else regularCalendarItemFontSize
+        var textVerticalPadding =
+            if (eventHeight <= shortCalendarItemHeight) shortCalendarItemVerticalPadding else regularCalendarItemVerticalPadding
+        textVerticalPadding = kotlin.math.min(eventHeight - fontSize, textVerticalPadding)
+        val textHorizontalPadding =
+            if (eventHeight <= shortCalendarItemHeight) shortCalendarItemHorizontalPadding else regularCalendarItemHorizontalPadding
```

Comment:
```suggestion
        val isShort = eventHeight <= shortCalendarItemHeight
        val fontSize = if (isShort) shortCalendarItemFontSize else regularCalendarItemFontSize
        var textVerticalPadding = if (isShort) shortCalendarItemVerticalPadding else regularCalendarItemVerticalPadding
        textVerticalPadding = kotlin.math.min(eventHeight - fontSize, textVerticalPadding)
        val textHorizontalPadding = if (isShort) shortCalendarItemHorizontalPadding else regularCalendarItemHorizontalPadding
```

Just a small suggestion in the name of DRY. But maybe you could come up with some data class that holds all of these params and you could construct it with just one if statement 🤔